metamorph/library/message_data_extractor.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '0.1'}
+
+DOCUMENTATION = '''
+---
+module: message_data_extractor
+short_description: Get important metadata from CI-message.
+version_added: 2.3.0
+author: Jiri Kulda (@Jurisak)
+options:
+  ci-message:
+    description:
+      - Input CI message in json format.
+    required: true
+
+  output:
+    description:
+      - Output metadata file name where CI Message metadata will be stored.
+    required: false
+    default: metamorph.json
+'''
+
+EXAMPLES = '''
+- name: Get metadata from CI-message
+  message_data_extractor:
+    ci-message: path_to/ci-message.json
+  register: result
+'''
+
+RETURN = '''
+Error message:
+  description: An error occurred during metadata extraction
+  returned: Error
+  type: dictionary
+  sample:  { "changed": false, "failed": true, "module_stderr": Given CI_message does not contain
+                                                                important key values. Missing key value: 'header'"
+CI-message metadata:
+  description: Extracted CI-message metadata
+  returned: changed
+  type: dictionary
+  sample: { "changed": true, "meta": { "messages": { "owner": "jkulda",
+                                                     "package": "setup",
+                                                     "release": "5.el7_1",
+                                                     "scratch": "true",
+                                                     "target": "rhel-*-*",
+                                                     "version": "2.8.71"}}
+'''
+
+from metamorph.lib.support_functions import setup_logging, read_json_file, storing_pretty_json
+import logging
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+class CIMessageKeyValueException(Exception):
+    pass
+
+
+class CIMessageReadingException(Exception):
+    pass
+
+
+class MessageDataExtractor(object):
+    """
+    Metadata Extractor from CI messages
+    """
+
+    def __init__(self, ci_message_file):
+        self.ci_message = {}
+        self.ci_message_file = ci_message_file
+        self.error_message = ""
+
+    def get_ci_message_data(self):
+        """
+        Main function which runs extractor behavior
+
+        :return Json with extracted metadata
+        """
+        self.read_input_file()
+        try:
+            if not self.check_valid_ci_message():
+                raise CIMessageKeyValueException("Given CI_message does not contain important data.")
+        except KeyError as key_detail:
+            raise CIMessageKeyValueException("Given CI_message does not contain important key values. "
+                                             "Missing key value: {}".format(key_detail))
+        return self.get_build_data()
+
+    def read_input_file(self):
+        try:
+            self.ci_message = read_json_file(self.ci_message_file)
+        except Exception as detail:
```

Comment:
Generally, I try to avoid such a wide Exception catch as the only one. 
It might be better to catch exactly what you're looking for. In this case, probably FileNotFoundError, and then probably something like json.decoder.JSONDecodeError.
```